### DISCUSSION TITLE
Is it possible to automate role creation or customize the use of lanes?

### DISCUSSION THREAD

**Discussion by mzabala23** [flowsforapex/apex-flowsforapex#441] [Q&A] [answered]
Hello,

Great product, thanks very much for that,
We have a question about lines, we have a custom authentication in apex app, the end user add a new user and a new rol (is it is necesary) we want to implement FLOWS4APEX but the lane always depend of roles created in Application Access Control and the API of APEX_ACL is not able to create new roles, so, is it possible to make use of lanes with custom authentication? more explicity custom roles

Thanks for your answer!

**Comment by rallen2010** [comment]
Hi,  Good question.  
So as you have figured out, in v22.1 (actually since about v4), our task assignment rules are quite simple...
1.  If you use BPMN Lanes in your diagram, you can amend the Task. Inbox View to map APEX Roles to BPMN Lanes, and shows tasks to potential users (members of the role) only.
2.  Users can reserve a task before they complete it to make sure that someone else is not working on the same task.

That's a little too simple for most real apps.  I build a demo app last year where I used Process Variables to keep track of who the possible users were for the next task.  And Variable expressions allow you to query from the database or run other logic to figure out who the task should be assigned to.  That's quite flexible and quite powerful, but not as Declarative as we would like.

We have a feature that we hope to develop in our next release that will re-do Task Assignment so that it is much more powerful, and you will be able to specify a particular user to be assigned to the task, or a list of potential users who can reserve the task, or a group of roles that could reserve the task.  This should make these problems much easier, and thereby simplify both your Flows for APEX models and your applications.

Hope this helps!

**Reply by mzabala23** [reply]
Hi Rallen, 
Thanks for your answer, we wait for the last version 23 we are very ansious with that release, but we have a question about the actual version, we have a table that store users, paswords and roles and one custom package that we use for authentication in our app, is possible to use that custom authentication with FLOWS4APEX and which is the correct way?

Thanks very much

**Reply by rallen2010** [reply]
Sorry for delay in responding. If you have done something custom, but know who the user is and what groups they are in, you should be able to custom- adapt the information in the task list to your table, and make a decision on it.  You might have to write a little bit of code to do it- but nothing too difficult, and we can probably give you some advice if you need.  The task list view contains info for the current tasks containing 
- Reserved (assigned) user
- Potential users
- Potential groups
- denied users
And from this you can enforce the policy you want a you have the user and group data.

**Comment by rallen2010** [comment, marked as answer]
Release 23.1, just released, should make Task Assignment very much easier - especially if you are using a recent APEX version with the APEX Task Inbox.
